From dc7a52faa27a2908d3db138fcad2bb1431943c9f Mon Sep 17 00:00:00 2001
From: Guillaume Aubertin <guillaume.aubertin@intel.com>
Date: Wed, 29 Oct 2014 13:58:50 +0100
Subject: [PATCH 1/3] bcmdhd: Makefile modifications to build several modules

bcmdhd driver can't support SDIO and PCIe in a single binary.
As we may have to support both bus with a single defconfig, Modifying the
Makefile to enable building a secondary module, as an out of tree module
with its specific config given as variables in the build command.

Change-Id: I0ef0e7cbd395265cff682ddcbc9dbc79370cc7cb
Signed-off-by: Guillaume Aubertin <guillaume.aubertin@intel.com>
---
 drivers/net/wireless/bcmdhd/Makefile | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/Makefile b/drivers/net/wireless/bcmdhd/Makefile
index 0a393b079e7b..d6e02f8bbaba 100644
--- a/drivers/net/wireless/bcmdhd/Makefile
+++ b/drivers/net/wireless/bcmdhd/Makefile
@@ -232,8 +232,24 @@ ifneq ($(CONFIG_BCMDHD_PCIE),)
   DHDOFILES += pcie_core.o
 endif
 
-bcmdhd-objs := $(DHDOFILES)
-obj-$(CONFIG_BCMDHD)   += bcmdhd.o
+# if the driver is built as part of the kernel build, we keep bcmdhd.ko name as usual.
+# in addition, it can also be built as an out of tree module with a different
+# name and different configuration flags passed through the make command line.
+
+ifeq ($(KBUILD_EXTMOD),)
+	bcmdhd-objs := $(DHDOFILES)
+	obj-$(CONFIG_BCMDHD)   += bcmdhd.o
+else
+ifneq ($(CONFIG_BCMDHD_PCIE),)
+	bcmdhd_pcie-objs := $(DHDOFILES)
+	obj-$(CONFIG_BCMDHD)   += bcmdhd_pcie.o
+endif
+ifneq ($(CONFIG_BCMDHD_SDIO),)
+	bcmdhd_sdio-objs := $(DHDOFILES)
+	obj-$(CONFIG_BCMDHD)   += bcmdhd_sdio.o
+endif
+endif
+
 
 all:
 	@echo "$(MAKE) --no-print-directory -C $(KDIR) SUBDIRS=$(CURDIR) modules"
-- 
2.0.3

